.github/workflows/update_chart_dependencies.yaml

```diff
@@ -31,10 +31,12 @@ jobs:
             component: 'operator'
             yaml_file_path: 'helm-charts/splunk-otel-collector/Chart.yaml'
             dependency_name: 'cert-manager'
-          - name: 'operator'
-            component: 'operator'
-            yaml_file_path: 'helm-charts/splunk-otel-collector/Chart.yaml'
-            dependency_name: 'opentelemetry-operator'
+          # TODO: Add this back once this chart has been updated to adhere to
+          # the operator chart schema validation rules. See: https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1065
```

Comment:
What extra properties do we supply?

Reply:
The upstream operator chart doesn't have a field to enable adding default instrumentation anymore since https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1065 was added.  This affected us and other companies. 
- Our use of `instrumentation.*` values fails validation tests
- Planning on working with the upstream to add default instrumentation to the operator chart 
  - Upstream added default instrumentation to the upstream kube-stack chart https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1078
  - We are likely going to have to align a little with upstream for enabling default instrumentation

Reply:
> doesn't have a field to enable adding default instrumentation

Which field?

Reply:
I cannot find `spec` section in https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-operator/values.yaml. Is this only used on our side?

Reply:
We should've not used a section that is supposed to be passed to the sub-chart for our own purposes. Now we should either:
1. move `instrumentation` out from `operator` breaking users relying on this option
2. or change the `operator` alias breaking user passing stuff to the operator.

The second option seems preferable to me. Similar to what you do, but instead of removing subchart, remove the alias

Reply:
- Operator Chart: Has no `instrumentation.*` field and the validation rules now enforce no additional fields.
- Splunk Chart: Yeah this is only used on our side with the `operator.instrumentation.*` field in our chart.  We are not the only company who took this approach, [sumo logic](https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/4a1d227d06595a20294ce51da225b01d0c81bf1c/deploy/helm/sumologic/values.yaml#L2450) and I'm sure other companies did too.

Reply:
I understand. But this is wrong

Reply:
Let's not drop the subchart. Just remove the alias

Reply:
Trying to explain why it's wrong: Imagine the sub-chart adds support of the `instrumentation` section with conflicting behavior.

Reply:
Actually, I think we should move `instrumentation` out of `operator`. Otherwise we break `operator.enabled` and `operator.admissionWebhooks`.

Reply:
- Yeah option 1 is starting to sound good to me. Going to think on it a little more. I wanted to handle updating our instrumentation helm values as part of a separate ticket.
- I should also mention the related blocker for this PR. The operator replaced several instrumentation feature gates with command-line arguments incrementally from version 0.95.0 (used before this PR) to version 0.99.0 (used after this PR). This is why I want to upgrade to operator v0.99.0 in this PR. Otherwise, our documentation would have to include feature gates and CLI arguments at the same time which is more confusing.

Reply:
This ticket is blocked until we complete following in a separate PR.
- Complete option 1, move operator.instrumentation out of the operator sub-chart values.
- Upgrade the operator sub-chart to the latest version
- This will include moving some code from this PR to the separate PR.